### PR TITLE
Game logic

### PR DESCRIPTION
**Pull request by edemagbenyo**
Implement game logic using various classes such as:
Game
Board
Player
Move

### CODE REVIEW

File: bin/main.rb [hunk 1]
```diff
@@ -1,2 +1,72 @@
...
+#Ask player 2 name
+#Checking for same name, if same ask player 2 for another name
+#Assign symbol to player 2
+#Display an empty board
+#Ask users to press a key to start game
+#Asking the user for their move by position 1-9

+#Prompt player 1 for move
+#Prompt player 2 for move
+#Display winner and name or draw
```
Comment (whiz25) [inline bin/main.rb]: remove these comments.

File: bin/main.rb [hunk 2]
```diff
@@ -1,2 +1,72 @@
...
+game = Game.new
+board.display
+
+current_player = player1
+while true
+  
+  #
+  puts "#{current_player.name} Enter your move, any number within 1-9 and press Enter. \nEnter 'q' or 'quit' to quit game. \Enter r to Restart the game"
+  # ask for input
+  # get input
```
Comment (whiz25) [inline bin/main.rb]: remove all comments from your code.

File: lib/move.rb [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+      puts 'Invalid move. Try another'
+      choice = gets.chomp
+    end
+    index = move_to_index(choice)
+
+    # if position is taken
+
+    while position_taken?(board, index)
+      puts 'Position taken try again.'
+      choice = gets.chomp
```
Comment (whiz25) [inline lib/move.rb]: this should be implemented in the ```main.rb``` file.

File: lib/board.rb [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+      false
+    end
+  end
+
+  def display
+    puts " #{@arr[0]} | #{@arr[1]} | #{@arr[2]} "
+    puts '-----------'
+    puts " #{@arr[3]} | #{@arr[4]} | #{@arr[5]} "
+    puts '-----------'
+    puts " #{@arr[6]} | #{@arr[7]} | #{@arr[8]} "
```
Comment (whiz25) [inline lib/board.rb]: this file should not have any ```puts, print``` but instead ```return``` statements.

**Review by whiz25 — CHANGES_REQUESTED** [review]
## Good job 👍 
You guys have met all the minimum requirements for this milestone. I am happy for you.
However, we need to fix a few things before we can merge this PR.
- [x] look at my comments carefully and implement the requested changes.

#### Make the requested changes and re-submit for another review.

### Martin Afani

**Comment by edemagbenyo** [comment]
Thanks Martin. We will implement the requested changes and submit another review.

**Review by whiz25 — APPROVED** [review]
You have implemented the requested changes and it's, therefore, a :+1: from me. Go ahead and merge.
### Martin Afani

**Pull request merged**